packages/react/src/components/MessageBar/MessageBar.types.ts

```diff
@@ -95,6 +95,12 @@ export interface IMessageBarProps extends React.HTMLAttributes<HTMLElement>, Rea
    * If unset, default will be the icon set by messageBarType.
    */
   messageBarIconProps?: IIconProps;
+
+  /**
+   * Custom role to apply to the MessageBar.
+   * @defaultvalue `alert` or `status` (based on `MessageBarType`)
+   */
+  role?: string;
```

Comment:
I'd suggest tightening this to `'alert' | 'status' | 'none'`. I don't think we want to let people apply literally any role to an internal div within a component, and I can't think of a use case for any other non-live-region role.

Reply:
Agreed, good point about not needing to support all possible roles here.

Reply:
Good call. This is done now: https://github.com/microsoft/fluentui/pull/18042/commits/3c7a219feeccd30537d9fbac3fbd50d55f44de31